juju_verify/verifiers/ceph.py

```diff
@@ -109,60 +137,83 @@ def __hash__(self) -> int:
         """Return hash representation of AZ objects."""
         return hash(self.__str__())
 
-    def find_parent(self, node_id: int) -> Optional[NodeInfo]:
-        """Find the node that has the id in the children list."""
+    def get_node(self, name: str) -> NodeInfo:
+        """Get node by name."""
+        if name not in self._nodes_name_map.keys():
+            raise KeyError(f"Node {name} was not found.")
+
+        try:
+            node = self._nodes[self._nodes_name_map[name]]
+            assert node.name == name  # check that variable _nodes was not changed
+        except (IndexError, AssertionError) as error:
+            raise ValueError("Private value `_nodes` was changed.") from error
+
+        return node
```

Comment:
Should this be called within the try statement? It would look more readable that if the try fails, an exception will be raised. Otherwise, the retrieved value will be returned.